File: 343. Integer Break.java
class Solution {
    public int integerBreak(int n) {
        if(n==3 || n==2){
            return n-1;
        }
        
        if(n%3==0){
            return (int) Math.pow(3, n/3);
        }
        else if (n % 3 == 1) {
            return (int) Math.pow(3, (n - 4) / 3) * 4;
        }
        else {
            return (int) Math.pow(3, (n / 3)) * 2;
        }
    }
    
}

File: 343. Integer Break.py
class Solution:
    def integerBreak(self, n: int) -> int:
        if (n==3 or n==2):
            return n-1

        if n%3==0:
            return 3**(n//3)
        elif n%3==1:
            return 3**((n-4)//3)*4
        else:
            return 3**(n//3)*2
